File: QuickBooksAPI/DataAccessLayer/Models/Customer.cs
namespace QuickBooksAPI.DataAccessLayer.Models
{
    public class Customer
    {
        public int Id { get; set; } // Local PK
        public string QboId { get; set; }
        public string UserId { get; set; }
        public string RealmId { get; set; }
        public string SyncToken { get; set; }
        public string Title { get; set; }
        public string GivenName { get; set; }
        public string  MiddleName { get; set; }
        public string FamilyName { get; set; }
        public string DisplayName { get; set; }
        public string CompanyName { get; set; }
        public bool Active { get; set; }
        public decimal Balance { get; set; }
        public string PrimaryEmailAddr { get; set; }
        public string PrimaryPhone { get; set; }
        public string BillAddrLine1 { get; set; }
        public string BillAddrCity { get; set; }
        public string BillAddrPostalCode { get; set; }
        public string BillAddrCountrySubDivisionCode { get; set; }
        public DateTime CreateTime { get; set; }
        public DateTime LastUpdatedTime { get; set; }
        public string Domain { get; set; }
        public bool Sparse { get; set; }
    }
}

File: QuickBooksAPI Frontend/app/src/types/customer.ts
export interface Customer {
  id: number;
  qboId: string;
  userId: string;
  realmId: string;
  syncToken: string;
  title: string;
  givenName: string;
  middleName: string;
  familyName: string;
  displayName: string;
  companyName: string;
  active: boolean;
  balance: number;
  primaryEmailAddr: string;
  primaryPhone: string;
  billAddrLine1: string;
  billAddrCity: string;
  billAddrPostalCode: string;
  billAddrCountrySubDivisionCode: string;
  createTime: string;
  lastUpdatedTime: string;
  domain: string;
  sparse: boolean;
}

export interface CreateCustomerRequest {
  givenName: string;
  middleName?: string;
  familyName: string;
  title?: string;
  suffix?: string;
  displayName: string;
  fullyQualifiedName?: string;
  companyName?: string;
  notes?: string;
  primaryEmailAddr?: { address: string };
  primaryPhone?: { freeFormNumber: string };
  billAddr?: {
    line1: string;
    city: string;
    countrySubDivisionCode: string;
    postalCode: string;
    country?: string;
  };
}

export interface UpdateCustomerRequest {
  id: string;
  syncToken: string;
  sparse?: boolean;
  displayName?: string;
  givenName?: string;
  familyName?: string;
  companyName?: string;
  primaryEmailAddr?: { address: string };
  primaryPhone?: { freeFormNumber: string };
  billAddr?: {
    line1: string;
    city: string;
    countrySubDivisionCode: string;
    postalCode: string;
    country?: string;
  };
  active?: boolean;
  balance?: number;
}

export interface DeleteCustomerRequest {
  id: string;
  syncToken: string;
  sparse: boolean;
  active: boolean;
}
